build/golang.go

```diff
@@ -130,11 +128,14 @@ func (gm *GoModule) getPackageZipLocation(cachePath, dependencyName, version str
 		return zipPath, nil
 	}
 
-	return gm.getPackagePathIfExists(filepath.Dir(cachePath), dependencyName, version)
+	return gm.getPackagePathIfExists(filepath.Dir(cachePath), encodedDependencyId)
 }
 
 // Validates that the package zip file exists and returns its path.
-func (gm *GoModule) getPackagePathIfExists(cachePath, dependencyName, version string) (zipPath string, err error) {
+func (gm *GoModule) getPackagePathIfExists(cachePath, encodedDependencyId string) (zipPath string, err error) {
+	moduleInfo := strings.Split(encodedDependencyId, ":")
```

Comment:
Although this is unlikely to even happen, let's return an error with a clear message, in case the length of moduleInfo isn't 2.